### ISSUE TITLE
prefs.logging.{save_script,file_log} without effect

### ISSUE THREAD

**Issue by schmitts** [brian-team/brian2#1407]
Hi!

I've noticed that `prefs.logging.save_script` and `prefs.logging.file_log` are without effect.

```python
from brian2 import prefs
prefs.logging.file_log = False

from brian2 import BrianLogger
print(BrianLogger.tmp_log)
'/tmp/brian_debug_t9ewkc80.log'
```

I suspect this is because the import of `prefs` also imports `brian2/__init__.py` in which the logger is initialized before the user has the chance to set any preferences.

Is my suspicion correct or do I do something wrong?

Thanks,

Sebastian

```
pip list
Package           Version
----------------- -------
asttokens         2.0.5
backcall          0.2.0
Brian2            2.5.0.3
Cython            0.29.28
decorator         5.1.1
executing         0.8.3
ipython           8.2.0
jedi              0.18.1
Jinja2            3.1.1
MarkupSafe        2.1.1
matplotlib-inline 0.1.3
mpmath            1.2.1
numpy             1.22.3
parso             0.8.3
pexpect           4.8.0
pickleshare       0.7.5
pip               22.0.4
pkg_resources     0.0.0
prompt-toolkit    3.0.29
ptyprocess        0.7.0
pure-eval         0.2.2
Pygments          2.12.0
pyparsing         3.0.8
setuptools        44.0.0
six               1.16.0
stack-data        0.2.0
sympy             1.10.1
traitlets         5.1.1
wcwidth           0.2.5
wheel             0.34.2
```

**Comment by mstimberg** [comment]
This is unfortunately correct, using these preferences only works via a preference file, see the warning here: https://brian2.readthedocs.io/en/stable/advanced/logging.html#preferences Preference files are explained here: https://brian2.readthedocs.io/en/stable/advanced/preferences.html#preference-files
In a way this would be addressed by #1217, but the general fix would involve to make the preferences notify other parts of the code of changes (e.g. via something like [traits](https://pypi.org/project/traits/)), or – easier to implement but a bit messier overall – make things like logging functions check whether the preferences changed since they were last used.

**Comment by schmitts** [comment]
Thanks. Sorry for missing the information in the documentation.
